### PR TITLE
Feature/like

### PR DESCRIPTION
**Pull request by IhorKalyta**

### CODE REVIEW

File: client/src/components/Post/index.tsx [hunk 1]
```diff
@@ -47,6 +53,37 @@ export default class Post extends React.Component<IProps> {
...
+    }
+
+    public onAddLike = (): void => {
+        const body = {userId: this.props.user._id, postId: this.props.userPosts.selectedPost._id};
+        this.props.userPosts.selectedPost.authorsOfLike.push(body.userId);
+        this.props.addLike(body);
+    }
+
+    public setLikesCount = (): boolean | void => {
+        if (this.props.userPosts.selectedPost.authorsOfLike !== undefined) {
```
Comment (InnaKutsik) [inline client/src/components/Post/index.tsx]: Достатньо -```!! This.props.userPosts.selectedPosts.authorsPost.authorsOfLikes```

File: client/src/components/Post/index.tsx [hunk 2]
```diff
@@ -47,6 +53,37 @@ export default class Post extends React.Component<IProps> {
...
+            this.props.setCountOfLikes(this.props.userPosts.selectedPost.authorsOfLike.length);
+
+            const arr = this.props.userPosts.selectedPost.authorsOfLike.filter((userId: string) => {
+                return this.props.user._id === userId;
```
Comment (InnaKutsik) [inline client/src/components/Post/index.tsx]: No need return in arrow function

Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: fixed

File: client/src/components/Post/index.tsx (hunk 1 above)
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: fixed

File: client/src/components/Post/index.tsx [hunk 3]
```diff
@@ -47,6 +53,37 @@ export default class Post extends React.Component<IProps> {
         this.props.getPostsAsync(this.props.user.username);
     }
 
+    public onDeleteLike = (): void => {
+        const body = {userId: this.props.user._id, postId: this.props.userPosts.selectedPost._id};
```
[the remark this answers is not in the document]
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: Yes i can but after demo.We have no time.

File: client/src/components/Post/index.tsx [hunk 4]
```diff
@@ -47,6 +53,37 @@ export default class Post extends React.Component<IProps> {
...
+        if (this.props.userPosts.selectedPost.authorsOfLike !== undefined) {
+            this.props.setCountOfLikes(this.props.userPosts.selectedPost.authorsOfLike.length);
+
+            const arr = this.props.userPosts.selectedPost.authorsOfLike.filter((userId: string) => {
+                return this.props.user._id === userId;
+            });
+
+            if (arr.length) {
+                this.props.checkUserLikeExist(true);
+                return true;
```
[the remark this answers is not in the document]
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: Yes we need.

File: client/src/components/Post/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: fixed

File: client/src/components/Post/index.tsx [hunk 5]
```diff
@@ -142,32 +182,35 @@ export default class Post extends React.Component<IProps> {
...
-                                            <span>72 likes</span>
-                                        </div>
-                                        <div className='d-lg-block d-none'>
-                                            <hr/>
-                                        </div>
-                                        <div className='mt-3 d-flex justify-content-between'>
+                                    </div>
+                                    <div className='d-lg-block d-none mt-1'>
+                                        {this.setLikesCount()}
+                                        {this.props.userPosts.selectedPost.authorsOfLike !== undefined &&
```
[the remark this answers is not in the document]
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: fixed

File: client/src/components/Post/index.tsx [hunk 6]
```diff
@@ -14,7 +14,13 @@ interface IProps {
     getPostsAsync: (username: string) => void;
     getMorePostsAsync: (username: string, page: number) => void;
     deletePhoto: () => void;
+    addLike: (body: {}) => void;
```
[the remark this answers is not in the document]
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: fixed

File: client/src/components/Post/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (IhorKalyta) [reply client/src/components/Post/index.tsx]: fixed

**Pull request merged**